src/main/java/com/demo/folder/service/TraineeService.java

```diff
@@ -1,132 +1,483 @@
+    String[] arr = new String[2];
+    arr[0] = plainTextPassword;
+    // Create Trainee and User
+    Trainee trainee = new Trainee();
+    User user = new User();
+
+    user.setFirstName(traineeRequestDTO.getFirstName());
+    user.setLastName(traineeRequestDTO.getLastName());
+    user.setUsername(Generator.generateUserName(traineeRequestDTO.getFirstName(),
+        traineeRequestDTO.getLastName()));
+    user.setPassword(passwordEncoder.encode(plainTextPassword));
+    user.setActive(true);
+
+    // Set the user to trainee
+    trainee.setUser(user);
+    trainee.setDateOfBirth(traineeRequestDTO.getDateOfBirth());
+    trainee.setAddress(traineeRequestDTO.getAddress());
+    createTrainee(trainee);
+    arr[1] = user.getUsername();
+    return arr;
+  }
+
+  @Transactional
+  public UpdateTraineeTrainersResponseDTO updateTraineeTrainersAdd(String traineeUserName,
+      UpdateTraineeTrainersRequestDTO requestDTO) {
+    Trainee trainee = traineeRepository.findByUsername(traineeUserName);
+    if (trainee == null) {
+      throw new IllegalArgumentException(
+          "Trainee with username " + traineeUserName + " not found.");
+    }
+    List<Trainer> trainers = new ArrayList<>();
+    for (String trainerUsername : requestDTO.getTrainerUsernames()) {
+      Trainer trainer = trainerRepository.findByUsername(trainerUsername);
+      if (trainer == null) {
+        throw new IllegalArgumentException(
+            "Trainer with username " + trainerUsername + " not found.");
+      }
+      trainers.add(trainer);
+      trainee.getTrainers().add(trainer);
+      trainer.getTrainees().add(trainee);
+      updateTrainee(trainee);
+      trainerService.updateTrainer(trainer);
+    }
+
+    return getUpdateTraineeTrainersResponseDTO(
+        trainers);
+  }
+
+  @Transactional
+  public UpdateTraineeTrainersResponseDTO updateTraineeTrainersRemove(String traineeUserName,
+      UpdateTraineeTrainersRequestDTO requestDTO) {
+    Trainee trainee = traineeRepository.findByUsername(traineeUserName);
+    if (trainee == null) {
+      throw new IllegalArgumentException(
+          "Trainee with username " + traineeUserName + " not found.");
+    }
+    List<Trainer> trainers = new ArrayList<>();
+    for (String trainerUsername : requestDTO.getTrainerUsernames()) {
+      Trainer trainer = trainerRepository.findByUsername(trainerUsername);
+      if (trainer == null) {
+        throw new IllegalArgumentException(
+            "Trainer with username " + trainerUsername + " not found.");
+      }
+      trainers.add(trainer);
+      trainee.getTrainers().remove(trainer);
+      trainer.getTrainees().remove(trainee);
+      updateTrainee(trainee);
+      trainerService.updateTrainer(trainer);
+    }
+
+    return getUpdateTraineeTrainersResponseDTO(
+        trainers);
+  }
+
+  private static UpdateTraineeTrainersResponseDTO getUpdateTraineeTrainersResponseDTO(
+      List<Trainer> trainers) {
+    return null;
```

Comment:
I think this method should be removed and the methods that call it should be made void (updateTraineeTrainersRemove(), updateTraineeTrainersAdd())